xtask/src/publish.rs

```diff
@@ -9,6 +9,22 @@ impl flags::PublishReleaseNotes {
     pub(crate) fn run(self, sh: &Shell) -> anyhow::Result<()> {
         let asciidoc = sh.read_file(&self.changelog)?;
         let mut markdown = notes::convert_asciidoc_to_markdown(std::io::Cursor::new(&asciidoc))?;
+
+        // Remove changelog header because GitHub misinterprets the changelog number as issue reference
+        let changelog_header_prefix = "# Changelog #";
+        let mut changelog_header_start =
+            markdown.find(changelog_header_prefix).expect("should contain changelog header");
+        changelog_header_start =
```

Comment:
This seems a little too complicated, and I'm not sure why it's needed.

My suggestion is to edit the AsciiDoc source, not the derived Markdown, since the latter is less predictable. That way, we only need to trim off the first line (until the first `'\n'`).

We could also strip off the `Commit: ...` line because it's redundant, but that seems less important.

Reply:
Good point; but it seems the AsciiDoc to Markdown conversion currently expects a title, and otherwise fails with "Error: document title not found". And I did not want to remove that code.

So I have changed it now to only remove the `#` from the changelog number. But since this deviates a bit from what this pull request was originally about, so I have marked it as Draft for now again. What do you think?

Reply:
Sorry, I didn't realize the title was needed for AsciiDoc.

I'd still prefer to strip off the line, since it's useless there.

Reply:
Maybe the changelog number is a bit useful though? For example when someone refers to it because they read it on the project website. I don't think GitHub itself numbers the releases.

In case you still think the complete title should be removed, should I then also remove the code which tries to convert this AsciiDoc title to Markdown? Or how should I handle the title being required currently?

(Also, in case it is easier for you to directly take this over, just let me know; this PR has "Allow edits by maintainers" enabled in case that is helpful.)